crates/assessor/src/lib.rs

```diff
@@ -82,14 +82,14 @@ pub struct AssessorInput {
 }
 
 impl AssessorInput {
-    /// Serializes the AssessorInput to a Vec<u8> using postcard.
-    pub fn to_vec(&self) -> Vec<u8> {
-        let bytes = postcard::to_allocvec(self).unwrap();
-        let length = bytes.len() as u32;
-        let mut result = Vec::with_capacity(4 + bytes.len());
-        result.extend_from_slice(&length.to_le_bytes());
-        result.extend_from_slice(&bytes);
-        result
+    /// Serialize the [AssessorInput] to a bytes vector.
+    pub fn encode(&self) -> Vec<u8> {
+        postcard::to_allocvec(&self).unwrap()
+    }
+
+    /// Deserialize the [AssessorInput] from a slice of bytes.
+    pub fn decode(bytes: &[u8]) -> Result<Self, impl core::error::Error> {
```

Comment:
Having this error type be `impl Error` might make this a bit harder to maintain over time and avoid breaking changes. No strong opinion here, but having it be a concrete type (and hide the inner error type if you want to avoid breaking changes in future) seems like a better option.

Curious the motivation, because if you instead intended for it to be the postcard error type, why not specify that directly?

Reply:
Good call, keeping me honest here. I've added errors created with `thiserror` in 796f4915b6b723d68b2ac7d7283cea8a39e59f9e and use this over anyhow everywhere in this library now.